test/State.js

```diff
@@ -266,6 +266,23 @@ describe('State', function() {
 		assert.strictEqual(1, valueFn.callCount);
 	});
 
+	it('should pass value and key args to validator function', function() {
+		var state = new State();
+		var keyName = 'key1';
+		var value = 2;
+		state.addToState({
+			[keyName]: {
+				validator: function(val, name) {
+					assert.strictEqual(val, value);
+					assert.strictEqual(name, keyName);
+
+					done();
```

Comment:
I think that the validator is not being called in this test case, because `done` is not being defined, so it would throw an error if this was called.

We only set a value for the first time after it's first used, so the validator will only run if you access it inside the test, doing `state.key` for example.